shuup/core/order_creator/_source.py

```diff
@@ -246,6 +248,25 @@ def status(self):
         if self.status_id:
             return OrderStatus.objects.get(pk=self.status_id)
 
+    @property
+    def language(self):
+        lang = settings.LANGUAGE_CODE
+        if self._language:
+            lang = self._language
+        elif self.customer:
+            lang = self.customer.language
+
+        if not is_existing_language(lang):
```

Comment:
I was thinking this too, but source.language can be set to something strange thus must be validated. imo.